narwhals/_dask/expr.py

```diff
@@ -490,6 +490,44 @@ class DaskExprStringNamespace:
     def __init__(self, expr: DaskExpr) -> None:
         self._expr = expr
 
+    def replace(
+        self,
+        pattern: str,
+        value: str,
+        *,
+        literal: bool = False,
+        n: int = 1,
+    ) -> DaskExpr:
+        return self._expr._from_call(
+            lambda _input, _pattern, _value, _literal, _n: _input.str.replace(
+                _pattern, _value, n=n, regex=not literal
+            ),
+            "replace",
+            pattern,
+            value,
+            n,
+            literal,
+            returns_scalar=False,
```

Comment:
As a general principle, we are not binding the arguments in the lambda, but passing those over:

```suggestion
            lambda _input, _pattern, _value, _literal, _n: _input.str.replace(
                _pattern, _value, regex=not _literal, n=_n
            ),
            "replace",
            pattern,
            value,
            literal,
            n,
            returns_scalar=False,
```

I caught this because the order of the args below were different from the lambda signature yet tests didn't fail

Reply:
Thaks for checking this 🙏